src/test/java/io/vertx/junit5/CountingCheckpointTest.java

```diff
@@ -101,4 +103,43 @@ void check_strict_checkpoint() {
       .isInstanceOf(IllegalStateException.class)
       .hasMessage("Strict checkpoint flagged too many times");
   }
+
+  @Test
+  @EnabledIfEnvironmentVariable(named = "SLOW_UNIT_TESTS", matches = ".*")
+  @DisplayName("Check of a lax checkpoint with max value")
+  void check_lax_checkpoint_max_value() {
+    AtomicLong satisfaction = new AtomicLong();
+    CountingCheckpoint checkpoint = CountingCheckpoint.laxCountingCheckpoint(
+        s -> satisfaction.incrementAndGet(), 1);
+
+    assertThat(satisfaction).hasValue(0);
+    checkpoint.flag();
+    assertThat(satisfaction).hasValue(1);
+    for (long i = 0; i < 2 * (long) Integer.MAX_VALUE + 2; i++) {
```

Comment:
I am not sure you need that many iterations, since the invariant is that `satisfaction` has value `1`.  Anyway since it's a lax checkpoint, you may just modify the implementation so that it stops counting once satisfied.